ts/DataGrid/DataGrid2/DataGridCell.ts

```diff
@@ -0,0 +1,154 @@
+/* *
+ *
+ *  Data Grid class
+ *
+ *  (c) 2020-2024 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Dawid Dragula
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import DataGridColumn from './DataGridColumn';
+import DataGridRow from './DataGridRow';
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Represents a cell in the data grid.
+ */
+class DataGridCell {
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    /**
+     * The HTML element of the cell.
+     */
+    public htmlElement: HTMLTableCellElement;
+
+    /**
+     * The column of the cell.
+     */
+    public column: DataGridColumn;
+
+    /**
+     * The row of the cell.
+     */
+    public row: DataGridRow;
+
+
+    /* *
+    *
+    *  Constructor
+    *
+    * */
+
+    /**
+     * Constructs a cell in the data grid.
+     *
+     * @param column The column of the cell.
+     * @param row The row of the cell.
+     */
+    constructor(column: DataGridColumn, row: DataGridRow) {
+        this.htmlElement = document.createElement('td');
+
+        this.column = column;
+        this.column.registerCell(this);
+
+        this.row = row;
+        this.row.registerCell(this);
+
+        this.htmlElement.addEventListener(
+            'mouseenter', this.onMouseEnter.bind(this)
+        );
+        this.htmlElement.addEventListener(
+            'mouseout', this.onMouseOut.bind(this)
+        );
+    }
+
+
+    /* *
+    *
+    *  Methods
+    *
+    * */
+
+    /**
+     * Renders the cell.
+     */
+    public render(): void {
+        if (!this.column.data) {
+            return;
+        }
+
+        const cellData = this.column.data[this.row.index];
+
+        this.htmlElement.innerText = '' + cellData;
+        this.row.htmlElement.appendChild(this.htmlElement);
+    }
+
+    /**
+     * Reflows the cell dimensions.
+     */
+    public reflow(): void {
+        this.htmlElement.style.width = this.column.getWidth() + 'px';
+    }
+
+    private onMouseEnter(): void {
```

Comment:
@vazonik : CSS looks kind of ugly, but just a heads up that the visuals here can be accomplished with pure CSS: https://codepen.io/davesantos/pen/mdZmyq. In theory.

Not saying it's a good/valid approach, and we might need those events for other stuff as well, but if not perhaps performance is better?

Reply:
Thanks for your comment!

All colors are as befits a PoC - clear, but not very pretty. :D Styling with the proper design will be done at the end, we still have to make different colors for light & dark theme.

> and we might need those events for other stuff as well

Yes, these events will be needed later for synchronization (such as highlight sync). Fortunately, we have virtual scrolling, so the number of event listeners is limited to the currently displayed cells.

Reply:
> All colors are as befits a PoC - clear, but not very pretty

When I said CSS is ugly I ment the example I referred to ;-)